hr_contract_multi_job/models/hr_contract.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2014 Savoir-faire Linux. All Rights Reserved.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    contract_job_ids = fields.One2many(
+        'hr.contract.job',
+        'contract_id',
+        string='Jobs')
+
+    # Modify the job_id field so that it points to the main job
+    job_id = fields.Many2one(
+        'hr.job',
+        string="Job Title",
+        compute="_compute_main_job_position",
+        store=True)
+
+    @api.depends('contract_job_ids')
+    @api.multi
+    def _compute_main_job_position(self):
+        """
+        Get the main job position from the field contract_job_ids which
+        contains one and only one record with field is_main_job == True
+        """
+        for contract in self:
+            main_job = contract.contract_job_ids.filtered('is_main_job')
+            if main_job:
+                main_job = main_job[0].job_id.id
```

Comment:
According to the `api.constraint` just below, I don't think this `if` statement is required. That's because:
- We are sure there is at least 1 `main_job`, which means `if main_job` is always True;
- We are also sure there is exactly 1 `main_job`, therefore `main_job` is a recordset containing exactly 1 record, which means its fields can be directly accessed through dot-notation (no need to use `[0]` to access 1st and only element)
- If you don't explicitely invoke a `write` method, you don't necessarily need to pass the ID (an integer), you can directly pass the record.

I'd rather write it as:

```
            main_job = contract.contract_job_ids.filtered('is_main_job').job_id
            contract.job_id = main_job
```
Or even write the same thing in a single line (if you manage to stay within the max line length)

Reply:
@nikul-serpentcs I argue that the condition is never false, because in the code there is a constraint ensuring there is always 1 and only 1 `main_job` . Am I wrong?